src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,192 @@
+    for (int i = 1; i < points.length; i++) {
+      index[i] = index[i - 1] + Math.sqrt(points[i - 1].distance(points[i]));
     }
-    if (shape instanceof QuadCurve2D) {
-      QuadCurve2D.Double curve = (QuadCurve2D.Double) shape;
-      curve = new QuadCurve2D.Double(curve.x1 + translateX, curve.y1 + translateY, curve.ctrlx + translateX,
-          curve.ctrly + translateY, curve.x2 + translateX, curve.y2 + translateY);
-      graphics.draw(curve);
+    return index;
+  }
+
+  private static void drawLineWithTranslate(Graphics2D graphics, Line2D line, double translateX, double translateY,
+      double scale) {
+    final Point2D point1 =
+        new Point2D.Double((line.getP1().getX() - translateX) * scale, (line.getP1().getY() - translateY) * scale);
+    final Point2D point2 =
+        new Point2D.Double((line.getP2().getX() - translateX) * scale, (line.getP2().getY() - translateY) * scale);
+    graphics.draw(new Line2D.Double(point1, point2));
+  }
+
+  protected static Point[] getRoutePoints(RouteDescription routeDescription, MapData mapData) {
+    final List<Territory> territories = routeDescription.getRoute().getAllTerritories();
+    final int numTerritories = territories.size();
+    final Point[] points = new Point[numTerritories];
+    for (int i = 0; i < numTerritories; i++) {
+      points[i] = mapData.getCenter(territories.get(i));
     }
+    if (routeDescription.getStart() != null) {
+      points[0] = routeDescription.getStart();
+    }
+    if (routeDescription.getEnd() != null && numTerritories > 1) {
+      points[numTerritories - 1] = new Point(routeDescription.getEnd());
+    }
+    return points;
   }
 
-  /**
-   * (x,y) - the first point to draw from (xx, yy) - the point to draw too
-   * (xxx, yyy) - the next point that the line segment will be drawn to
-   */
-  private static void drawCurvedLineWithNextPoint(final double x, final double y,
-      final double xx, final double yy, final double xxx, final double yyy, final List<Shape> shapes) {
-    final int maxControlLength = 150;
-    double controlDiffx = xx - xxx;
-    double controlDiffy = yy - yyy;
-    if (Math.abs(controlDiffx) > maxControlLength || Math.abs(controlDiffy) > maxControlLength) {
-      double ratio = 0.0;
-      try {
-        ratio = Math.abs(controlDiffx / controlDiffy);
-      } catch (final ArithmeticException ex) {
-        ratio = 1000;
-      }
-      if (Math.abs(controlDiffx) > Math.abs(controlDiffy)) {
-        controlDiffx = controlDiffx < 0 ? -maxControlLength : maxControlLength;
-        controlDiffy = controlDiffy < 0 ? (int) (-maxControlLength / ratio) : (int) (maxControlLength / ratio);
-      } else {
-        controlDiffy = controlDiffy < 0 ? -maxControlLength : maxControlLength;
-        controlDiffx = controlDiffx < 0 ? (int) (-maxControlLength * ratio) : (int) (maxControlLength * ratio);
-      }
+  protected static Tuple<double[], double[]> pointsToDoubleArrays(Point[] points) {
```

Comment:
I recommend avoiding using `Tuple`. Plus there is some redundancy in the code below. I think it can be fixed up with a helper method having this API:

`private static double[] getValues(Point[] point, java.function.Function<Point, Double> extractor)`

And you could call it like this:
```
double[] xValues = getValues(points, point -> point.getX())
double[] yValues = getValues(points, point -> point.getY())
```